src/databricks/labs/ucx/assessment/aws.py

```diff
@@ -107,11 +110,50 @@ def list_attached_policies_in_role(self, role_name: str):
             attached_policies.append(policy.get("PolicyArn"))
         return attached_policies
 
+    def list_all_uc_roles(self):
+        list_roles_cmd = (
+            f"iam list-roles --profile {self._profile}"
+        )
+        roles = self._run_json_command(list_roles_cmd)
+        uc_roles = []
+        for role in roles["Roles"]:
+            try:
+                policy_document = role["AssumeRolePolicyDocument"]
+                for statement in policy_document["Statement"]:
+                    if statement["Effect"] != "Allow":
+                        continue
+                    if statement["Action"] != "sts:AssumeRole":
+                        continue
+                    principal = statement["Principal"].get("AWS")
+                    if not principal:
+                        continue
+                    if isinstance(principal, list):
+                        is_uc_principal = False
+                        for single_principal in principal:
+                            if single_principal in self.UC_MASTER_ROLES_ARN:
+                                is_uc_principal = True
+                                continue
+                        if not is_uc_principal:
+                            continue
+                    elif principal not in self.UC_MASTER_ROLES_ARN:
+                        continue
+                    uc_roles.append(
+                        AWSRole(
+                            role_id=role.get("RoleId"),
+                            role_name=role.get("RoleName"),
+                            arn=role.get("Arn"),
+                            path=role.get("Path")
+                        )
+                    )
+            except KeyError:
```

Comment:
Don't. Check if key is in dictionary instead.

Reply:
Fixed